src/main/scala/mesosphere/marathon/core/task/tracker/TaskTracker.scala

```diff
@@ -35,5 +35,69 @@ trait TaskTracker {
 }
 
 object TaskTracker {
-  case class App(appName: PathId, tasks: Iterable[MarathonTask])
+  /**
+    * Contains all tasks grouped by app ID.
+    */
+  case class AppDataMap private (appTasks: Map[PathId, TaskTracker.App]) {
```

Comment:
I guess it is outside of the scope of this PR, but it would be nice at some point to rename `AppDataMap` to `TasksByApp`.